pkg/trace/stats/concentrator.go

```diff
@@ -46,8 +47,25 @@ type Concentrator struct {
 	agentEnv               string
 	agentHostname          string
 	agentVersion           string
-	peerSvcAggregation     bool // flag to enable peer.service aggregation
-	computeStatsBySpanKind bool // flag to enable computation of stats through checking the span.kind field
+	peerSvcAggregation     bool     // flag to enable peer.service aggregation
+	computeStatsBySpanKind bool     // flag to enable computation of stats through checking the span.kind field
+	peerTagKeys            []string // keys for supplementary tags that describe peer.service entities
+}
+
+func prepareTagKeys(tags ...string) []string {
+	if len(tags) == 0 {
+		return nil
+	}
+	ptm := make(map[string]struct{})
```

Comment:
Why a `struct{}`? A `bool` would probably be easier here, and is more common.

You can also avoid looping over all of these twice by doing something like this:
```
// Remove duplicates
var deduped []string
seen := make(map[string]bool)
for _, tag := range tags {
  if !seen[tag] {
    seen[tag] = true
    deduped = append(deduped, tag)
  }
}
```

Reply:
Interesting, I learned that `map[string]struct{}` was more idiomatic for this thing a while back?
I'm not opposed to using bool though.

The suggestion to dedupe in the same first pass is good, will do.

Reply:
In datadog's own backend, it's much more common to use a `struct{}` as it uses less space I believe. I like the idea of modifying both datastructures in a single loop.